gtfu/src/main/java/gtfu/tools/DayLogSlicer.java

```diff
@@ -59,15 +63,21 @@ public DayLogSlicer(TripCollection tripCollection, RouteCollection routeCollecti
             agentMap.put(tripID,agent);
 
             String latLon = String.valueOf(lat) + String.valueOf(lon);
+
             // If there is no LatLonMap for this trip, add it.
             if(gpsMap.get(tripID) == null){
                 Map<String, GPSData> latLonMap = new HashMap();
                 gpsMap.put(tripID,latLonMap);
             }
-            // If there is no existing GPSData for this latLon value, add it
+
+            // If there is no existing GPSData for this latLon value, add it and update previousUpdate for that trip
             if (gpsMap.get(tripID).get(latLon) == null) {
-                gpsMap.get(tripID).put(latLon, new GPSData(seconds * 1000l, lat, lon));
+                // previousUpdateMap stores the most recent GPS update timestamp (in seconds) for each trip_id.
+                // it relies on the list being sorted by timestmap, which it is.
+                gpsMap.get(tripID).put(latLon, new GPSData(seconds * 1000l, ((previousUpdateMap.get(tripID) != null) ? seconds - previousUpdateMap.get(tripID) : -1), lat, lon));
```

Comment:
I could see an argument for being too complex of a line - can calculate this previous update time in a previous line.

Reply:
Yeah, that's pretty ugly.